### PR TITLE
LoginRequiredMiddleware unittests

### PR DESCRIPTION
**Pull request by PetrushynskyiOleksii**
## JIRA

* [PYTHON-197](https://ssu-jira.softserveinc.com/browse/PYTHON-197)

## Code reviewers

- [ ] @meelros
- [ ] @BoroviyOrest
- [ ] @Kardan1123
- [ ] @mizin4ik
- [ ] @dimigor
- [ ] @Sasha1152

### Second Level Review

- [ ] @lhalam

## Summary of change

1. Created tests for custom `LoginRequired` middleware.
2. Changed response value to constants from `responsehelpers` module in `LoginRequiredMiddleware`.

## CHECK LIST
- [x]  СI passed
- [x]  Сode coverage >=95%
- [x]  PR is reviewed manually again (to make sure you have 100% ready code)
- [x]  All reviewers agreed to merge the PR
- [x]  I've checked new feature as logged in and logged out user if needed
- [x]  PR meets all conventions

### CODE REVIEW

**Review by BoroviyOrest — DISMISSED** [review]

File: way_to_home/tests/unittests/middlewares/testloginrequired.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+            response = self.user_client.put(path, data, content_type='application/json')
+            self.assertEqual(response.status_code, 403)
+
+            response = self.user_client.patch(path, data, content_type='application/json')
+            self.assertEqual(response.status_code, 403)
+
+    def test_forbidden_request_for_guest(self):
+        """Provide tests unsuccessful requests for guest."""
+        path = '/api/v1/place/'
```
Comment (dimigor) [inline way_to_home/tests/unittests/middlewares/testloginrequired.py]: use `reverse('place')`

File: way_to_home/tests/unittests/middlewares/testloginrequired.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+        response = self.guest_client.put(path, data, content_type='application/json')
+        self.assertEqual(response.status_code, 403)
+
+        response = self.guest_client.patch(path, data, content_type='application/json')
+        self.assertEqual(response.status_code, 403)
+
+    def test_success_request_for_guest(self):
+        """Provide tests successful requests for guest."""
+        path = '/api/v1/user/login'
```
Comment (dimigor) [inline way_to_home/tests/unittests/middlewares/testloginrequired.py]: The same situation as above
if our API version changes, we will not need to change this path
So rather use `reverse('login_user')`

File: way_to_home/tests/unittests/middlewares/testloginrequired.py (hunk 1 above)
Comment (PetrushynskyiOleksii) [reply way_to_home/tests/unittests/middlewares/testloginrequired.py]: Thank you, I forgot about it :)

File: way_to_home/tests/unittests/middlewares/testloginrequired.py (hunk 2 above)
Comment (PetrushynskyiOleksii) [reply way_to_home/tests/unittests/middlewares/testloginrequired.py]: Yes, done.

**Review by BoroviyOrest — DISMISSED** [review]

**Review by victoria-kit — DISMISSED** [review]

File: way_to_home/tests/unittests/middlewares/testloginrequired.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+from middlewares.login_required import GUESTS_PATHS
+
+
+class LoginRequiredTestCase(TestCase):
+    """TestCase for providing LoginRequired middleware testing."""
+
+    def setUp(self):
+        """Provide preparation before testing middleware."""
+        user = CustomUser(id=100, email='testuser@mail.com', is_active=True)
+        user.set_password('testpassword')
```
Comment (MaxDubchak) [inline way_to_home/tests/unittests/middlewares/testloginrequired.py]: Password like this doesn't pass Password validation

File: way_to_home/tests/unittests/middlewares/testloginrequired.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    def test_put_with_invalid_json(self):
+        """Provide tests for PUT request with invalid JSON data."""
+        path = '/api/'
+        invalid_json = 'invalid json'
+
+        response = self.client.put(path, invalid_json, content_type='application/json')
+        self.assertEqual(response.status_code, 400)
+
+    def test_forbidden_request_for_user(self):
+        """Provide tests for user`s GET requests with paths that available only for guests."""
```
Comment (MaxDubchak) [inline way_to_home/tests/unittests/middlewares/testloginrequired.py]: """Provide tests for user`s GET requests with paths that **are** available only for guests."""

File: way_to_home/tests/unittests/middlewares/testloginrequired.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        for path in GUESTS_PATHS:
+            response = self.user_client.get(path)
+            self.assertEqual(response.status_code, 403)
+
+    def test_forbidden_request_for_guest(self):
+        """Provide tests for guest`s requests with path that available only for users."""
```
Comment (MaxDubchak) [inline way_to_home/tests/unittests/middlewares/testloginrequired.py]: """Provide tests for user`s GET requests with paths that **are** available only for guests."""

File: way_to_home/tests/unittests/middlewares/testloginrequired.py (hunk 3 above)
Comment (PetrushynskyiOleksii) [reply way_to_home/tests/unittests/middlewares/testloginrequired.py]: I think it is not so important :)
But I changed it.

**Review by Sasha1152 — DISMISSED** [review]
Good!

**Review by BoroviyOrest — APPROVED** [review]

**Review by dimigor — APPROVED** [review]

**Pull request merged**